File: freelancerprojectmanager.client/src/Model/Project.ts
import type { ProjectStatus } from "./ProjectStatus";
import type { PTask } from "./PTask";

export interface Project {
    estimateValue?: number;
    taskCount: number;
    id: number;
    name: string;
    description: string;
    client?: Client;
    startDate?: Date;
    endDate?: Date;
    status: ProjectStatus;
    loggedMinutes?: number;
    /**
     * available when fetching a single project, not available when fetching all projects
     */
    tasks?:PTask[];
}



export interface Client {
    id: string;
    name: string;
}

File: FreelancerProjectManager.Server/Domain/ProjectManagement/Project.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace FreelancerProjectManager.Server.Domain.ProjectManagement
{
    public enum ProjectStatus
    {
        Scoping,
        Active,
        OnHold,
        Completed,
        Canceled
    }
    public class Project 
    {
        
        public int ID { get; set; }
        public ProjectStatus Status { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        [ForeignKey(nameof(Client))]
        public int ClientID { get; set; }
        public Client Client { get;set; }
        public List<PTask> Tasks { get; set; }
        public DateTime CreatedAt { get; set; }

        public int LoggedMinutes { get; set; }
        public decimal? EstimateValue { get; set; }

        public decimal? GetActualHourlyRate()
        {
            if(EstimateValue==null || EstimateValue<=0 || LoggedMinutes<=0)
            {
                return null;
            }
            return (EstimateValue.Value/(LoggedMinutes/60m));
        }

    }
}
